quickstart/docker/image/add-router.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+# For edge router setup we will need to address some external variables.
+# this section needs to move to doucmentation.
+#export EXTERNAL_IP="$(curl -s eth0.me)"       
+#export ZITI_EDGE_CONTROLLER_IP_OVERRIDE="127.0.0.1"
+#export ZITI_EDGE_ROUTER_IP_OVERRIDE="${EXTERNAL_IP}"
+#export ZITI_EDGE_CONTROLLER_PORT=1280
+#export ZITI_EDGE_CONTROLLER_MANAGEMENT_PORT=6262
+#export ZITI_EDGE_ROUTER_PORT=443
+#export ZITI_EDGE_ROUTER_LAN_PORT="ens3"
+#export ZITI_EDGE_ROUTER_NAME="ER_NEWNAME"
+
+
+# For ease of use, we will introduce an optional parameter for controller ip.
+if [ $# -eq 0 ]; then
+   if [[ "${ZITI_EDGE_CONTROLLER_IP_OVERRIDE}" != "" ]]; then
+      controller_ip=${ZITI_EDGE_CONTROLLER_IP_OVERRIDE}
+   else
+      controller_ip="127.0.0.1"
+   fi
+elif [ $# -eq 1 ]; then
+   # user specified controller ip
+   controller_ip=$1
+else
+   echo ""
+   echo "Usage:"
+   echo "     $0 <controller-ip>"
+   exit
+fi
+
+if [[ "${ZITI_EDGE_CONTROLLER_PORT}" != "" ]]; then
+   ctrl_port=${ZITI_EDGE_CONTROLLER_PORT}
+else
+   ctrl_port="1280"
+fi
+
+if [[ "${ZITI_EDGE_CONTROLLER_MANAGEMENT_PORT}" != "" ]]; then
+   mgmt_port=${ZITI_EDGE_CONTROLLER_MANAGEMENT_PORT}
+else
+   mgmt_port="6262"
+fi
+
+if [[ "${ZITI_EDGE_ROUTER_LAN_PORT}" != "" ]]; then
+   lan_port=${ZITI_EDGE_ROUTER_LAN_PORT}
+else
+   lan_port="ens3"
+fi
+
+if [[ "${ZITI_EDGE_ROUTER_PORT}" != "" ]]; then
+   rport=${ZITI_EDGE_ROUTER_PORT}
+else
+   rport="443"
+fi
+
+if [[ "${ZITI_EDGE_ROUTER_NAME}" != "" ]]; then
+   er_name=${ZITI_EDGE_ROUTER_NAME}
+else
+   er_name="none"
+fi
+
+router_ip=$(ip addr show ${lan_port} | grep "inet\b" | awk '{print $2}' | cut -d/ -f1)
+
+echo Use ${router_ip} as internal IP
+
+cat <<EOT >config.yml
```

Comment:
you should probably be using `ziti create config router edge|private` here instead of emitting the config file. see the ziti-cli-functions.sh for examples if interested like: https://github.com/openziti/ziti/blob/release-next/quickstart/docker/image/ziti-cli-functions.sh#L770

Reply:
yes, Geoff was able to add environment variable to set external IP to SANs (for ziti create config).  Also would need to have the external IP in the advertisement section. (I believe that is been worked on right now).